### PR TITLE
BH Static TLB changes for worker cores

### PR DESCRIPTION
**Pull request by pjanevskiTT**
Part of larger effort on Budabackend to bring up and test Blackhole.

This PR adds Blackhole specific functions for TLBs. It also removes assertions for usage of static TLBs for Blackhole, which were added during initial Blackhole bringup.

[Issue on BBE for core static TLBs](https://yyz-gitlab.local.tenstorrent.com/tenstorrent/budabackend/-/issues/2653)

### CODE REVIEW

File: device/tt_silicon_driver.cpp [hunk 1]
```diff
@@ -3063,8 +3059,21 @@ void *tt_SiliconDevice::channel_0_address(std::uint32_t offset, std::uint32_t de
...
+    // Temporary hack for blackhole bringup.
+    if (arch_name == tt::ARCH::BLACKHOLE) {
+
+        // Use 195th 2MB TLB onwards.
+        const std::uint32_t TLB_CH0_START = 195;
+        bar0_offset = offset - architecture_implementation->get_dram_channel_0_peer2peer_region_start()
+                        + architecture_implementation->get_dynamic_tlb_2m_base() + (TLB_CH0_START * architecture_implementation->get_dynamic_tlb_2m_size());
+
+    } else {
+        // This hard-codes that we use 16MB TLB #1 onwards for the mapping. See tt_SiliconDevice::init_pcie_tlb.
```
Comment (joelsmithTT) [inline device/tt_silicon_driver.cpp:3073]: Can you remove the duplicate comment at the top of the function?

Does `tt_SiliconDevice::init_pcie_tlb` still exist?

File: device/tt_silicon_driver.cpp [hunk 2]
```diff
@@ -3063,8 +3059,21 @@ void *tt_SiliconDevice::channel_0_address(std::uint32_t offset, std::uint32_t de
...
     auto architecture_implementation = pci_device->hdev->get_architecture_implementation();
-    std::uint64_t bar0_offset = offset - architecture_implementation->get_dram_channel_0_peer2peer_region_start()
-                                + architecture_implementation->get_dynamic_tlb_16m_base() + architecture_implementation->get_dynamic_tlb_16m_size();
+    std::uint64_t bar0_offset;
+
+    // Temporary hack for blackhole bringup.
+    if (arch_name == tt::ARCH::BLACKHOLE) {
+
+        // Use 195th 2MB TLB onwards.
+        const std::uint32_t TLB_CH0_START = 195;
```
Comment (abhullar-tt) [inline device/tt_silicon_driver.cpp:3068]: is there a specific reason that the 195th 2MB TLB is the start?

Comment (pjanevskiTT) [reply device/tt_silicon_driver.cpp:3067]: Added a comment, other TLBs before are used as either static of fallback

File: device/tt_silicon_driver.cpp (hunk 1 above)
Comment (pjanevskiTT) [reply device/tt_silicon_driver.cpp]: Removed, also removed part of the comment for `init_pcie_tlb`, it doesn't exist

**Review by broskoTT — APPROVED** [review]

**Pull request merged**